validator/src/main.rs

```diff
@@ -0,0 +1,261 @@
+use std::{
+    net::{IpAddr, SocketAddr},
+    process::exit,
+    sync::Arc,
+    time,
+};
+
+use anyhow::{anyhow, Context, Result};
+use clap::Parser;
+use rand::distributions::{Alphanumeric, DistString};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+    sync::Notify,
+};
+use tracing::{debug, error, info, Instrument};
+
+#[derive(Parser)]
+#[clap(version)]
+struct Args {
+    #[clap(
+        long,
+        env = "CNI_VALIDATOR_LOG_LEVEL",
+        default_value = "cni_validator=info,warn"
+    )]
+    log_level: kubert::LogFilter,
+
+    #[clap(long, default_value = "plain")]
+    log_format: kubert::LogFormat,
+
+    #[clap(long, default_value = "0.0.0.0:4140")]
+    outbound_proxy_addr: SocketAddr,
+
+    #[clap(parse(try_from_str), long, env = "KUBERNETES_SERVICE_HOST")]
+    target_ip: IpAddr,
+
+    #[clap(parse(try_from_str), long, env = "KUBERNETES_SERVICE_PORT")]
+    target_port: u16,
```

Comment:
tiny nit: this probably doesn't actually matter that much, since this CLI isn't actually going to be run by real humans, but it feels a tiny bit weird that we take the outbound proxy's address as an `{ip}:{port}` string, but we take the target IP and port as separate arguments? would it make more sense to make these more consistent with each other?

Reply:
Yeah, 100% agreed, it's super unintuitive. It's unfortunate k8s doesn't really expose an env variable that captures the whole target, and that's why the args were split. We can set the default from a different env variable here (such as `KUBERNETES_ADDR`) and configure that at the deployment level.

e.g we change to `target_addr` as an arg, and in the deployment:
```yaml
env
   - name: KUBERNETES_ADDR
      value: "$(KUBERNETES_SERVICE_HOST):$(KUBERNETES_SERVICE_PORT)"
```